command/agent/agent.go

```diff
@@ -184,12 +184,26 @@ func (c *cmd) run(args []string) int {
 	if config == nil {
 		return 1
 	}
+	// Check if we have a log file path or a data-dir path
+	// If we have a log path, we expect the file name to be present
+	// In case we have the data-dir, a default name is used.
+	var logFilePath string
+	if config.LogFile == "" && config.DataDir == "" {
+		logFilePath = ""
+	} else if config.LogFile != "" {
```

Comment:
I think we probably should default to no logging. Many people already run Consul in a way that redirects the stdout logging to syslog/journald and we wouldn't want to fill up their disk.